importer/src/main/java/org/cbio/portal/pipelines/importer/config/clinical/ClinicalDataReader.java

```diff
@@ -0,0 +1,245 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbio.portal.pipelines.importer.config.clinical;
+
+import org.cbio.portal.pipelines.importer.model.*;
+import org.cbio.portal.pipelines.importer.config.composite.CompositeClinicalData;
+
+import java.util.*;
+import org.apache.commons.logging.*;
+
+import org.springframework.batch.item.*;
+import org.springframework.batch.item.file.FlatFileItemReader;
+import org.springframework.batch.item.file.mapping.*;
+import org.springframework.batch.item.file.transform.*;
+
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.validation.BindException;
+
+/**
+ *
+ * @author ochoaa
+ */
+public class ClinicalDataReader implements ItemStreamReader<CompositeClinicalData> {
+    
+    // params from job exeuction context
+    private CancerStudy cancerStudy;
+    private Map<String, List<ClinicalAttribute>> dataFileClinicalAttributes;
+    
+    private List<CompositeClinicalData> compositeClinicalDataResults = new ArrayList();
+    
+    private static final Log LOG = LogFactory.getLog(ClinicalDataReader.class);    
+    
+    public static enum MissingAttributeValues {
+        NOT_APPLICABLE("Not Applicable"),
+        NOT_AVAILABLE("Not Available"),
+        PENDING("Pending"),
+        DISCREPANCY("Discrepancy"),
+        COMPLETED("Completed"),
+        NULL("null"),
+        MISSING(""),
+        NA("NA"),
+        N_A("N/A"),
+        UNKNOWN("unknown");
+
+        private String propertyName;
+        
+        MissingAttributeValues(String propertyName) { this.propertyName = propertyName; }
+        public String toString() { return propertyName; }
+
+        static public boolean has(String value) {
+            if (value == null) return false;
+            if (value.trim().equals("")) return true;
+            try { 
+                value = value.replaceAll("[\\[|\\]\\/]", "");
+                value = value.replaceAll(" ", "_");
+                return valueOf(value.toUpperCase()) != null;
+            }
+            catch (IllegalArgumentException ex) {
+                return false;
+            }
+        }
+
+        static public String getNotAvailable() {
+            return "[" + NOT_AVAILABLE.toString() + "]";
+        }
+    }
+    
+    @Override
+    public void open(ExecutionContext executionContext) throws ItemStreamException {
+        this.cancerStudy = (CancerStudy) executionContext.get("cancerStudy");        
```

Comment:
@n1zea144 here's where I get the cancer study object from the execution context for clinical data